### DISCUSSION TITLE
How to use webhooks with Inventree?

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#4409] [Q&A]
I've not a coder by any stretch of the imagination but know enough to hack my way around  to get something working. 

I've got a small website that sells electronic parts (which I've assembed)  on eBay and my own website (Woocommerce) 
I have hacked together a python script to keep the stock in sync between the 2) 

I would now like to integrate this into Inventree. Reducing the stock count on inventree seems simple enough using a API call. So for example when a item is sold on woocommere, a webhook is made to my website which then reduces the stock on eBay/inventree.

What I would like to do now is use webhooks on inventree so when a assembled part is built and the stock number increases then a webhook is done to my website which will update eBay etc. 

The problem is I've looked through the api docs for webhooks on inventree and no closer in figuring out how it works. 

Suppose the first question, is this possible yet on inventree? Im guessing so since I've seen some integration into Shopify.  So then how do I setup a webhook?

The only docs I can find are:
Webhooks
```
create
POST /api/webhook/{endpoint}/
Process incomming webhook.
```

Any ideas woud be welcome.

**Comment by miggland** [comment]
The webhook you mention is for making inventree have a webhook, which you could use to call Inventree from Woocommerce.

For the case when the event occurs in Inventree, you probably want the event plugin: https://docs.inventree.org/en/latest/extend/plugins/event/

**Reply by StormTrooper** [reply]
Thanks, that clears it up.  I'll have a look into plugins.

**Comment by matmair** [comment]
Hi there @StormTrooper we currently only have functions for receiving webhooks - sending them is not done yet (I have no real usage for it as InvenTree is my source of truth) - see  https://github.com/inventree/InvenTree/issues/2038

**Reply by StormTrooper** [reply]
Thanks for the clarrification. I might look into changing my source of truth as well  to make things easier.
